site/source/tutorials/005_form_widgets/index.md

```diff
@@ -1,26 +1,159 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:1-4,7-9,19-24,26-28,31,33,45-48,109-111 %}
+
+This widget will render an empty form with a `submit` handler that prevents the form from being submitted to the server. Before we continue to expand on this starting point though, let's get the form integrated into the application so that we can see the form develop as we add more features.
+
+To start, add the CSS rules that the widget is going to need. Include the following rules in the `workerForm.css` file.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/styles/workerForm.css' lang:css %}
+
+Now, let's add the `WorkerForm` to the `App` class. Import the class and update the `render` method to draw it. It should be included after the `Banner` and before the `WorkerContainer` so the `render` method should look like this:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:73,78-80,105-110 %}
+
+{% aside 'Reminder' %}
+If you can't see the application, remember to run `dojo build -w` to build the application and start the development server.
+{% endaside %}
+
+Now, open the application in a browser and inspect it via the browsers developer tools. Notice that the empty form element is being rendered onto the page as expected.
+
+## Adding form widgets
+Our form is going to contain fields to allow the user to enter the worker's first name, last name and e-mail address. It will also contain a save button that will use the form's data to create a new worker. We could create these fields and buttons using the `v` function to create simple virtual DOM elements. If we did this, however, we would have to handle details such as theming, internationalization ([i18n](https://en.wikipedia.org/wiki/Internationalization_and_localization)) and accessibility ([a11y](https://en.wikipedia.org/wiki/Accessibility)) ourselves. Instead, we are going to leverage some of Dojo 2's built-in widgets that have been designed with these things in mind. Add the following imports to `WorkerForm.ts`:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerForm.ts' lines:5-6 %}
+
+
+{% aside 'Note' %}
+	If you have been working with the same project throughout this series, you might get some errors when you try to import the widgets. If you do, add a new dependency to the `project.json` file: "@dojo/widgets":"2.0.0-alpha.23" and rerun `npm install` to install Dojo 2's standard widget library.
+{% endaside %}
+
+We are importing the `Button` class that will be used to provide the form's submit button and the `TextInput` class that will provide the data-entry fields for the worker data.
+
+Let's use those classes and a few virtual DOM nodes, to add the visual elements of the form:
+
+```ts
```

Comment:
Unfortunately, I can't. The "finished" version contains the event handlers as well and so source code for these lines would end with trailing commas.